### PR TITLE
Remove IsStream imports from FileSystem.File module

### PR DESCRIPTION
**Pull request by rnjtranjan**
Fixes:https://github.com/composewell/streamly/issues/1840

### CODE REVIEW

**Review by adithyaov — COMMENTED** [review]
Added a review commit

File: src/Streamly/Internal/FileSystem/File.hs [hunk 1]
```diff
@@ -99,19 +99,18 @@ import Streamly.Internal.Control.Concurrent (MonadAsync)
...
-import Streamly.Internal.Data.Stream.IsStream.Type (IsStream)
 import Streamly.Internal.Data.Unfold.Type (Unfold(..))
 -- import Streamly.String (encodeUtf8, decodeUtf8, foldLines)
 import Streamly.Internal.System.IO (defaultChunkSize)
 
 import qualified Streamly.Data.Array.Unboxed as A
 import qualified Streamly.Data.Unfold as UF
 import qualified Streamly.Internal.Data.Fold.Type as FL
-    (Step(..), snoc, initialize)
+    (Step(..), drain, snoc, initialize)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/File.hs:109]: Add drain this to the Streamly.Data.Fold import above.

File: src/Streamly/Internal/FileSystem/File.hs [hunk 2]
```diff
@@ -99,19 +99,18 @@ import Streamly.Internal.Control.Concurrent (MonadAsync)
 import Streamly.Internal.Data.Array.Unboxed.Type (Array(..), writeNUnsafe)
 import Streamly.Internal.Data.Fold.Type (Fold(..))
 import Streamly.Internal.Data.Stream.Type (Stream)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/File.hs:101]: We should import this from exposed module.

File: src/Streamly/Internal/FileSystem/File.hs [hunk 3]
```diff
@@ -99,19 +99,18 @@ import Streamly.Internal.Control.Concurrent (MonadAsync)
...
 import qualified Streamly.Data.Array.Unboxed as A
 import qualified Streamly.Data.Unfold as UF
 import qualified Streamly.Internal.Data.Fold.Type as FL
-    (Step(..), snoc, initialize)
+    (Step(..), drain, snoc, initialize)
 import qualified Streamly.Internal.Data.Unfold as UF (bracket)
 import qualified Streamly.Internal.FileSystem.Handle as FH
 import qualified Streamly.Internal.Data.Array.Stream.Foreign as AS
-import qualified Streamly.Internal.Data.Stream.IsStream as S
+import qualified Streamly.Internal.Data.Stream as S
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/File.hs:113]: Can we import the exposed module here? And import only those APIs from internal that are not exposed.

File: src/Streamly/Internal/FileSystem/File.hs [hunk 4]
```diff
@@ -99,19 +99,18 @@ import Streamly.Internal.Control.Concurrent (MonadAsync)
...
 -- import Streamly.String (encodeUtf8, decodeUtf8, foldLines)
 import Streamly.Internal.System.IO (defaultChunkSize)
 
 import qualified Streamly.Data.Array.Unboxed as A
 import qualified Streamly.Data.Unfold as UF
 import qualified Streamly.Internal.Data.Fold.Type as FL
-    (Step(..), snoc, initialize)
+    (Step(..), drain, snoc, initialize)
 import qualified Streamly.Internal.Data.Unfold as UF (bracket)
 import qualified Streamly.Internal.FileSystem.Handle as FH
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/File.hs:111]: Can we import exposed module and import only unexposed from internal?

**Pull request merged**